### PR TITLE
redirect cerr and cout

### PR DESCRIPTION
**Pull request by lalo**
It's a bit janky but throwing it out here just to judge usefulness.
It hides the help info to have a path for migration once logging has more knobs to configure.

This could enable merging vw_varinfo python script into the vw python api.

### CODE REVIEW

File: vowpalwabbit/global_data.h [hunk 1]
```diff
@@ -311,6 +312,28 @@ struct vw_logger
   vw_logger& operator=(const vw_logger& other) = delete;
 };
 
+struct buffer_restore
```
Comment (olgavrou) [inline vowpalwabbit/global_data.h:315]: `buffer_replace` instead of `buffer_restore`?

**Review by olgavrou — APPROVED** [review]

File: vowpalwabbit/parse_args.cc [hunk 2]
```diff
@@ -1350,6 +1350,16 @@ vw& parse_args(
   {
     time(&all.init_time);
 
+    std::string cerr_filename, cout_filename;
+    option_group_definition redirect_args("Redirect cerr/cout to filename");
+    redirect_args.hide().add(make_option("cerr", cerr_filename).help("filename to redirect cerr to"));
```
Comment (jackgerrits) [inline vowpalwabbit/parse_args.cc:1355]: `cerr_file`?

File: vowpalwabbit/parse_args.cc [hunk 3]
```diff
@@ -1350,6 +1350,16 @@ vw& parse_args(
...
+    redirect_args.hide().add(make_option("cout", cout_filename).help("filename to redirect cout to"));
```
Comment (jackgerrits) [inline vowpalwabbit/parse_args.cc:1356]: `cout_file`?

**Review by jackgerrits — APPROVED** [review]

File: vowpalwabbit/global_data.h (hunk 1 above)
Comment (lalo) [reply vowpalwabbit/global_data.h]: done

File: vowpalwabbit/parse_args.cc (hunk 2 above)
Comment (lalo) [reply vowpalwabbit/parse_args.cc]: done

**Pull request merged**